extension/src/XliffFunctions.ts

```diff
@@ -0,0 +1,1112 @@
+): number {
+  const matchMap: Map<string, string[]> = getXlfMatchMap(matchXlfDoc);
+  return matchTranslationsFromTranslationMap(
+    matchXlfDoc,
+    matchMap,
+    languageFunctionsSettings
+  );
+}
+
+export function importTranslatedFileIntoTargetXliff(
+  source: Xliff,
+  target: Xliff,
+  languageFunctionsSettings: LanguageFunctionsSettings,
+  translationFolderPath: string
+): void {
+  if (languageFunctionsSettings.translationMode !== TranslationMode.dts) {
+    throw new Error(
+      "The setting NAB.UseDTS is not active, this function cannot be executed."
+    );
+  }
+  const dictionary = getDictionary(
+    languageFunctionsSettings.useDTSDictionary,
+    target.targetLanguage,
+    translationFolderPath
+  );
+  source.transunit.forEach((sourceTransUnit) => {
+    let targetTransUnit = target.getTransUnitById(sourceTransUnit.id);
+    if (targetTransUnit === undefined) {
+      // a new translation
+      targetTransUnit = sourceTransUnit;
+      target.transunit.push(targetTransUnit);
+    } else {
+      if (!isTranslatedState(targetTransUnit.target.state)) {
+        if (targetTransUnit.targets.length === 0) {
+          // No target element
+          targetTransUnit.targets.push(sourceTransUnit.target);
+        } else {
+          if (
+            sourceTransUnit.target.stateQualifier === StateQualifier.idMatch
+          ) {
+            targetTransUnit.target.stateQualifier = undefined;
+          } else {
+            targetTransUnit.target.state = sourceTransUnit.target.state;
+            targetTransUnit.target.stateQualifier =
+              sourceTransUnit.target.stateQualifier;
+            targetTransUnit.target.textContent =
+              sourceTransUnit.target.textContent;
+          }
+        }
+      }
+    }
+    targetTransUnit.target.textContent = dictionary
+      ? dictionary.searchAndReplace(targetTransUnit.target.textContent)
+      : targetTransUnit.target.textContent;
+    changeStateForExactMatch(languageFunctionsSettings, targetTransUnit);
+    detectInvalidValues(targetTransUnit, languageFunctionsSettings);
+  });
+}
+
+function changeStateForExactMatch(
+  languageFunctionsSettings: LanguageFunctionsSettings,
+  targetTransUnit: TransUnit
+): void {
+  if (
+    languageFunctionsSettings.exactMatchState !== undefined &&
+    isExactMatch(targetTransUnit.target.stateQualifier)
+  ) {
+    targetTransUnit.target.state = languageFunctionsSettings.exactMatchState;
+    targetTransUnit.target.stateQualifier = undefined;
+  }
+}
+
+function isTranslatedState(state: TargetState | undefined | null): boolean {
+  return [
+    TargetState.translated,
+    TargetState.signedOff,
+    TargetState.final,
+  ].includes(state as TargetState);
+}
+
+function isExactMatch(stateQualifier: string | undefined): boolean {
+  return stateQualifier
+    ? [(StateQualifier.exactMatch, StateQualifier.msExactMatch)].includes(
+        stateQualifier as StateQualifier
+      )
+    : false;
```

Comment:
Wait, won't below give the same result?
```suggestion
  return [(StateQualifier.exactMatch, StateQualifier.msExactMatch)].includes(
        stateQualifier as StateQualifier
      );
```
undefined as StateQualifier = undefined, wasn't it?

Reply:
True! Nice find!

Reply:
Apparently it doesn't take me more 4 lines to forget :sweat_smile:

Reply:
done